_posts/algorithm-development/guides/2016-10-13-git-guide.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: article
+title:  "Using Git with Algorithmia"
+excerpt: "Access Algorithmia from Git"
+date: 2016-10-13  09:50:46
+permalink: /algorithm-development/git-support/
+tags: [algo-guide-git]
+show_related: false
+author: steph_kim
+image:
+    teaser: /icons/Algorithmia_Basics.png
+---
+
+Every algorithm on Algorithmia has a dedicated git repository. This allows you to seamlessly integrate Algorithmia with your current workflows,
+such as your code editor, IDE, or deployment pipelines. 
+
+<img src="/images/post_images/legit/legit-demo.gif" alt="Demo for Git" class="screenshot">
+
+## Accessing a repository
+
+You can clone existing algorithms from the following endpoint:
+{% highlight bash %}
+    git clone https://git.algorithmia.com/git/:username/:algoname.git
+{% endhighlight %}
+
+Replace the `:username` and `:algoname` portions with the currect values, case sensitive.
+
+## Committing changes to an algorithm
+
+You can apply your changes and push them as usual.
+{% highlight bash %}
+git commit -a -m "Note on changes goes here"
+git push
+{% endhighlight %}
+
+Everything will behave as a normal git repository, with the additional last line that will point you to a private REST API endpoint.
+
+{% highlight bash %}
+Remote: build successful for algo://:username/:algoname/:hash
+{% endhighlight %}
+
+The REST API endpoint is a live version of the latest build of your algorithm (previous versions are still available). It behaves exactly as a published algorithm 
+except for the fact that it is not listed in the marketplace and will only accept calls from your API keys. This is a great way 
+to run regression tests against an algorithm before making it available to others.
+
+## Testing your changes 
+
+With the private REST API endpoint in hand, you can run your unit tests against it or manually test that algorithm from the command 
+line using the <a href="https://github.com/algorithmiaio/algorithmia-cli">Algorithmia CLI</a>. Testing using the CLI will be
+something similar to:
+{% highlight bash %}
+algo run :username/:algoname/:hash -d "<JSON_INPUT>"
+{% endhighlight %}
+
+See the <a href="https://github.com/algorithmiaio/algorithmia-cli">Algorithmia CLI</a> for more options.
+
+## Managing branches
+
+Branches are allowed and will be accepted in pushes, however Algorithmia will not compile branches other than master.
+
+Algorithmia will only compile the head of the master branch on each push. It will not compile intermediate commits 
+if they are all pushed together as a batch. Finally, Algorithmia will reject any push which contains non-fast-forward commits. 
+This is to prevent changing the history of an algorithm, and potentially breaking Algorithmia's versioning.
+
+## Initializing a repository
+
+As of today, you will need to manually create an algorithm from the web interface, which will `git init` a repository. Once a 
```

Comment:
By listing the `git init` command here it is a little confusing ... is the user supposed to run that command? ... Better to leave it out